Cargo.toml

```diff
@@ -146,6 +146,7 @@ serde-xml-rs = "0.6.0"
 serde_json = "1.0.94" # update roc_std/Cargo.toml on change
 serial_test = "1.0.0"
 signal-hook = "0.3.15"
+smallvec = { version = "1.10.0", features = ["const_generics", "const_new"] }
```

Comment:
we already have `arrayvec` as a dependency. would that not work here?

Reply:
It looks like arrayvec will not spill onto the heap if the array gets filled up?

Reply:
Ah true, didn't see you use that here